src/rgw/rgw_iam_policy.h

```diff
@@ -371,6 +370,14 @@ struct Condition {
     }
   };
 
+  struct string_like : public std::binary_function<const std::string,
```

Comment:
good question, i hadn't seen that. looks like it's for `std::not2()` in `Condition::eval()` - the documentation for http://en.cppreference.com/w/cpp/utility/functional/not2 seems to suggest wrapping it in `std::function` as an alternative. or we could just add the `first_argument_type` and `second_argument_type` typedefs manually. what do you think @adamemerson?